src/main/kotlin/no/nav/dagpenger/regel/api/db/SubsumsjonStore.kt

```diff
@@ -23,24 +16,14 @@ interface SubsumsjonStore {
 
     fun insertBehov(behov: InternBehov): Int
     fun hentKoblingTilEkstern(eksternId: EksternId): BehandlingsId
-    fun getBehov(behovId: String): InternBehov
-    fun behovStatus(behovId: String): Status
+    fun getBehov(behovId: UlidId): InternBehov
+    fun behovStatus(behovId: UlidId): Status
     fun insertSubsumsjon(subsumsjon: Subsumsjon, created: ZonedDateTime = ZonedDateTime.now(ZoneId.of("UTC"))): Int
-    fun getSubsumsjon(behovId: String): Subsumsjon
-    fun getSubsumsjonByResult(subsumsjonId: SubsumsjonId): Subsumsjon
+    fun getSubsumsjon(behovId: UlidId): Subsumsjon
+    fun getSubsumsjonByResult(subsumsjonId: UlidId): Subsumsjon
     fun delete(subsumsjon: Subsumsjon)
 }
 
-data class SubsumsjonId(val id: String) {
```

Comment:
Det er veldig fint å skille på det forskjellige Id'ene så ville gjerne beholde denne

en ting som vil ha løst det, er på denne måten

```

data class SubsumsjonId(private val rawId: String) {

    val id: String

    init {
        try {
            id = ULID.parseULID(rawId).toString()
        } catch (e: IllegalArgumentException) {
            throw IllegalSubsumsjonIdException("Id $rawId is not a valid subsumsjon id")
        }
    }

}
```

Reply:
samme med BehovId

Reply:
` ULID.parseULID(rawId).toString()` gir alltid upper

Reply:
og lage samme for BehovId

```
data class BehovId(private val rawId: String) {

    val id: String

    init {
        try {
            id = ULID.parseULID(rawId).toString()
        } catch (e: IllegalArgumentException) {
            throw IllegalSubsumsjonIdException("Id $rawId is not a valid subsumsjon id")
        }
    }

}
```

Reply:
Eller! Med litt kotlin magi: 

```
data class Ulid(private val rawId: String) {

    val id: String

    init {
        try {
            id = ULID.parseULID(rawId).toString()
        } catch (e: IllegalArgumentException) {
            throw IllegalSubsumsjonIdException("Id $rawId is not a valid subsumsjon id")
        }
    }

}

typealias BehovId = Ulid
typealias SubsumsjonId = Ulid

```

Reply:
versjon 2:

```
data class Ulid(private val rawId: String) {

    val id: String

    init {
        try {
            id = ULID.parseULID(rawId).toString()
        } catch (e: IllegalArgumentException) {
            throw IllegalSubsumsjonIdException("Id $rawId is not a valid subsumsjon id")
        }
    }

    override fun equals(other: Any?): Boolean {
        return if(other is Ulid) {
            this.id == other.id
        } else {
            false
        }

    }



}
```

Reply:
````
data class Ulid(private val rawId: String) {

    val id: String

    init {
        try {
            id = ULID.parseULID(rawId).toString()
        } catch (e: IllegalArgumentException) {
            throw IllegalSubsumsjonIdException("Id $rawId is not a valid subsumsjon id")
        }
    }

    override fun equals(other: Any?): Boolean {
        return if(other is Ulid) {
            this.id == other.id
        } else {
            false
        }

    }



}
```

Reply:
Okei! Da skal vi være over på BehovId og SubsumsjonId = Ulid